controllers/machinedeletionremediation_controller.go

```diff
@@ -90,57 +90,68 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 		return ctrl.Result{}, err
 	}
 
-	annotations := remediation.GetAnnotations()
-	if annotations != nil {
-		machineNameNamespace, exists := annotations[MachineNameNamespaceAnnotation]
-		if exists {
-			machineName, machineNamespace, err := extractNameAndNamespace(machineNameNamespace, remediation.GetName())
-			if err != nil {
-				log.Error(err, "could not get Machine data from remediation", "remediation", remediation.GetName(), "annotation", machineNameNamespace)
-				return ctrl.Result{}, nil
-			}
+	log.Info("Machine Deletion Remediation CR found", "name", remediation.GetName())
 
-			ok, err := r.verifyMachineIsDeleted(ctx, machineName, machineNamespace, remediation.Name)
-			if err != nil {
+	// Get Machine's Name and Namespace to retrive it from the cluster
+	var (
+		machineName, machineNamespace string
+		gotMachineFromNode            bool
+	)
+
+	if remediation.GetAnnotations() != nil {
+		if machineNameNamespace, exists := remediation.GetAnnotations()[MachineNameNamespaceAnnotation]; exists {
+			if machineName, machineNamespace, err = extractNameAndNamespace(machineNameNamespace, remediation.GetName()); err != nil {
+				log.Error(err, "could not get Machine data from remediation", "remediation", remediation.GetName(), "annotation", machineNameNamespace)
 				return ctrl.Result{}, err
 			}
-			if !ok {
-				return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-			}
-			return ctrl.Result{}, nil
 		}
+	} else {
+		if node, err := r.getNodeFromMdr(remediation); err != nil {
+			return ctrl.Result{}, err
+		} else if machineName, machineNamespace, err = r.getMachineNameNamespaceFromNode(node); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if err = r.setDeletedMachineAnnotation(ctx, remediation, machineName, machineNamespace); err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "failed to update remediation CR with timeout annotation")
+		}
+		gotMachineFromNode = true
 	}
 
-	var node *v1.Node
-	if node, err = r.getNodeFromMdr(remediation); err != nil {
+	var machine *unstructured.Unstructured
+	if machine, err = r.getMachine(machineName, machineNamespace); err != nil {
+		if apiErrors.IsNotFound(err) && !gotMachineFromNode {
+			log.Info(successfulMachineDeletionInfo, "node", remediation.Name, "machine", machineName)
+			return ctrl.Result{}, nil
+		}
+		// If Machine's name and namespace come from a Node, the Machine should still exists
+		log.Error(err, noMachineFoundError, "node", remediation.Name, "machine", machineName)
```

Comment:
IIUC this log correspond to the use-case where apiErrors.IsNotFound(err) == true && gotMachineFromNode == true.
Does it make sence for apiErrors.IsNotFound(err) == false ?